api/scpca_portal/models/dataset.py

```diff
@@ -188,6 +215,22 @@ def original_files(self) -> Iterable[OriginalFile]:
     def original_file_paths(self) -> Set[Path]:
         return {Path(of.s3_key) for of in self.original_files}
 
+    def get_hash_data(self) -> int:
+        original_files_sorted = sorted(self.original_files, key=lambda of: of.s3_key)
+        concat_hash = "".join(of.hash for of in original_files_sorted)
+        return hash(concat_hash.strip())
+
+    def get_hash_metadata(self) -> int:
+        libraries_metadata = Library.get_libraries_metadata(self.libraries)
+        metadata_file_contents = metadata_file.get_file_contents(libraries_metadata)
+        return hash(metadata_file_contents)
+
+    def get_hash_readme(self) -> int:
```

Comment:
This looks right but we need to account for the `date` in the readme. That first line is going to make this not work day to day. For now I think we should just pop off the first line before calculating the hash.